### DISCUSSION TITLE
Set Product and Serial for ESP32-S2

### DISCUSSION THREAD

**Discussion by lambda-aw** [micropython/micropython#15317] [ESP32] [answered]
Hello,
I tried to compile ESP32 S2 port and it works nicelly, even without much experience, I can get it done. However, I am wondering if I can set Product and/or Serial with some build variables, so I can filter for specific boards when I have more of them connected to PC. if both are compile-time onstants I can live with Product only, but cannot find a way - both micropython and IDF are big beasts for newcomers.
Tried `idf.py menuconfig`, but no entry for it.
Thanks.

Jaroslav

**Comment by robert-hh** [comment, marked as answer]
You can for instance change MICROPY_HW_BOARD_NAME in the board's mpconfigboard.h. This is just a string. Drawback: It has to be re-compiled for every board.

**Reply by lambda-aw** [reply]
Thanks. It is not a problem, I can call the build in the loop and storing 10 files is not big deal, not optimal, but not much a burden.
Any chance to supply it on command line, or rather have `for` and `sed`?

**Reply by robert-hh** [reply]
Since it is a C define, you should be able to set it from the cmake script, and let that thae the value from command line. With make I would know how to do that. But cmake is unknown land.

**Reply by lambda-aw** [reply]
hm, still cannot figure out. when I flash the official uPy it gives me Product `Espressif Device` and Serial `123456`. I can change the `MICROPY_HW_BOARD_NAME`, but this is visible only inside the interpreter, not from the OS. I wonder, whether some changes into tinyusb could help somehow.

![image](https://github.com/micropython/micropython/assets/127849376/b77d947f-8a9f-47ad-83f4-90e48561a081)

**Comment by bixb922** [comment]
As for a serial number, there is the MAC address. I capture that on the PC with ```mpremote exec "import network;print(network.WLAN(network.STA_IF).config('mac'))"``` and use the result in shell scripts or makefiles to do different stuff for each board.

**Reply by lambda-aw** [reply]
that could work, I hoped for something without connecting micropython, but no way so far.
